awx/api/views.py

```diff
@@ -145,6 +145,15 @@ def destroy(self, request, *args, **kwargs):
         # Still allow deletion of new status, because these can be manually created
         if obj.status in ACTIVE_STATES and obj.status != 'new':
             raise PermissionDenied(detail=_("Cannot delete running job resource."))
+        elif not obj.events_processed:
+            # Prohibit deletion if job events are still coming in
+            if obj.modified + dateutil.relativedelta.relativedelta(minutes=1) < now():
```

Comment:
actually, maybe it would be better to use `obj.finished` here instead.

And @chrismeyersfsu I'm not really a fan of putting the timedelta logic inside of the model's method. It would make the field confusing to use. If the job has been finished for a long while and the events aren't fully processed, or if this is happening on the regular, that would be a bad state of the server, and it's better to show that as opposed to hiding it.